code/DMSDocumentCartExtension.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+/**
+ * Class DMSDocumentCartExtension adds a control
+ *
+ * @property Boolean     AllowedInCart
+ *
+ * @property DMSDocument owner
+ */
+class DMSDocumentCartExtension extends DataExtension
+{
+    private static $db = array(
+        'AllowedInCart' => 'Boolean',
+    );
+
+    public function isAllowedInCart()
```

Comment:
We could add the contract yes, but the user must still be able to persist to the DB whether a document is AllowedInCart or not.